File: TSOL.Domain/Entities/Candidate.cs


using System;

namespace TSOL.Domain.Entities
{
    public class Candidate : BaseEntity
    {
        public string UserName { get; set; }
        public string Password { get; set; }

        public string FullName { get; set; }
        public string Email { get; set; }
        public string PhoneNumber { get; set; }
        public string Address { get; set; }
        public DateTime ? Birthdate { get; set; }
         

        public bool Status { get; set; } = true;
        public DateTime ? CreatedDate { get; set; }  
        public DateTime ? LastActivateDate { get; set; }  
    }
}

File: testonline-ui/src/app/model/candidate.ts
import { Quiz } from './quiz';
import { Summary } from 'src/app/model/summary';

export class Candidate {
  id: number;
  index: number;
  userName: string;
  password: string;
  fullName: string;
  phoneNumber: string;
  address: string;
  email: string;
  status: number;
  createdDate: Date;
  lastActivateDate: Date;

  subjectNames: string [];

  finalScore: number;
  quizes: Quiz[];
  startTime: Date;
  submitTime: Date;
  summary: Summary;

  constructor(data: any) {
    data = data || {};
    this.id = data.id;
    this.index = data.index;
    this.userName = data.userName;
    this.password = data.password;
    this.subjectNames = data.subjectNames;
    this.phoneNumber = data.phoneNumber;
    this.address = data.address;
    this.status = data.status;
    this.createdDate = data.createdDate;
    this.lastActivateDate = data.lastActivateDate;
    this.finalScore = data.finalScore;
    this.quizes = data.quizes;
    this.startTime = data.startTime;
    this.submitTime = data.submitTime;
    this.summary = data.summary;
  }

}
